src/dawn/IIR/IIRNodeIterator.h

```diff
@@ -61,22 +80,32 @@ class IIRNodeIterator {
             IIRNodeIterator<typename RootIIRNode::ChildType, LeafNode>(iterator_->get());
       }
     }
+    // If the restIterator contains no children, we need to continue advancing the iterator until we
+    // fall into a node with children (or leaves) to evaluate or we reach the end
+    while(isTop_ && isVoidIter() && !isEnd())
```

Comment:
assume I have a stage with no domethod and I iteraten over the Stages of a Stencil. This would leave out the stage without domethod since it `isVoidIter()` right? Is this intentionally